src/certificate/commands/AdrCertificateCommand.ts

```diff
@@ -0,0 +1,65 @@
+import { Service } from 'typedi';
+import { ICertificatePayload } from '../../models';
+import { CERTIFICATE_DATA } from '../../models/Enums';
+import { TechRecordType } from '../../models/Types';
+import { TechRecordService } from '../../tech-record/TechRecordService';
+import { BasePayloadCommand } from '../ICertificatePayloadCommand';
+
+@Service()
+export class AdrCertificateCommand extends BasePayloadCommand {
+	constructor(private techRecordService: TechRecordService) {
+		super();
+	}
+
+	private certificateIsAnAdr = (): boolean => this.state.type === CERTIFICATE_DATA.ADR_DATA;
+
+	public async generate(): Promise<ICertificatePayload> {
+		if (!this.certificateIsAnAdr()) {
+			return {} as ICertificatePayload;
+		}
+
+		const { testResult } = this.state;
+
+		const adrDetails: TechRecordType<any> = await this.techRecordService.getAdrDetails(testResult);
+		const makeAndModel = await this.techRecordService.getVehicleMakeAndModel(testResult);
```

Comment:
As these appear non dependent, we could optimise this by utilising `Promise.all` to make these requests in parallel e.g.

```
const [adrDetails, makeAndModel] = await Promise.all([
  this.techRecordService.getAdrDetails(testResult),
  this.techRecordService.getVehicleMakeAndModel(testResult)
]);
```

Reply:
Good spot! Will update 👍